### PR TITLE
Add m3 proteins

### PR DESCRIPTION
**Pull request by fgrunewald**
Implement M3 support for disordered proteins and fix molmeta implementation as well as non-edges.

### CODE REVIEW

File: polyply/src/ff_parser_sub.py [hunk 1]
```diff
@@ -24,12 +24,10 @@ class PolyplyFFParser(FFDirector):
     '''
     @SectionLineParser.section_parser('moleculetype', 'edges',
                                       negate=False, context_type='block')
-    @SectionLineParser.section_parser('moleculetype', 'non-edges',
+    @SectionLineParser.section_parser('moleculetype', 'non-_parse_edges(tokens, context, context_type, negate)edges',
```
Comment (pckroon) [inline polyply/src/ff_parser_sub.py:27]: This looks like a typo

File: polyply/src/ff_parser_sub.py [hunk 2]
```diff
@@ -64,7 +70,7 @@ def _parse_edges_new(tokens, context, context_type, negate):
         atomname = prefixed_atom[0]
         if atomname not in context and context_type == 'modification':
             raise KeyError(error_message.format(atomname, context_type,
-                                                context.name))
+                                            context.name))
```
Comment (pckroon) [inline polyply/src/ff_parser_sub.py:73]: lint

File: polyply/tests/test_apply_links.py [hunk 3]
```diff
@@ -181,112 +181,243 @@ def test_match_link_and_residue_atoms_fail(example_meta_molecule,
...
+                                                        meta={"version": 1})],
+                          [vermouth.molecule.Interaction(atoms=(0, 1, 2, 3),
+                                                        parameters=['1', '0.33', '500'],
+                                                        meta={})]],
+                         [[], [[3 , {'atomname':'BB8' , 'order': 1}]]],
+                         [[], []],
+                         [[(1, 4, 1)],[(1, 4, 5, 6, 1)]],
+                         [[(0, 0)], [(1, 0)]]
+                        ),
+                        # 7 - test non-edges which fails
```
Comment (pckroon) [inline polyply/tests/test_apply_links.py:304]: Fails why?

File: polyply/tests/test_apply_links.py [hunk 4]
```diff
@@ -181,112 +181,243 @@ def test_match_link_and_residue_atoms_fail(example_meta_molecule,
...
-                                              link,
-                                              link_to_resid)
+        link.non_edges = non_edges
+        link.patterns = patterns
+        try:
+            processor.apply_link_between_residues(example_meta_molecule,
+                                                  link,
+                                                  link_to_resid)
+        except MatchError:
+            continue
```
Comment (pckroon) [inline polyply/tests/test_apply_links.py:409]: Shouldn't this cause a failing test? I.e. shouldn't non-matching links get their own test to make sure they don't match?

**Review by pckroon — CHANGES_REQUESTED** [review]
in test_apply_links it's unclear to me how/why those are non-edges, rather than normal bonds. Also, for the cases you indicate fail (6 and 8), specify why.

File: polyply/tests/test_apply_links.py (hunk 4 above)
Comment (fgrunewald) [reply polyply/tests/test_apply_links.py:409]: Yeah I guess we could do that as well

**Review by pckroon — APPROVED** [review]
Little bit of lint left here and there, but LGTM. Note that I didn't check all the data files in detail.
Also, is there a reason to redistribute the M3 ff files rather than taking them from vermouth?

**Comment by fgrunewald** [comment]
Concerning redistribution: 1) links die on the dihedralphase and there is no way to fix this, because polyply doesn't have coordinates. 2) the terminals are written slightly differently to avoid having to introduce a flag for setting the terminals. For now you just get the charged unless you making a single monomer.

**Pull request merged**